### DISCUSSION TITLE
Unable to run API

### DISCUSSION THREAD

**Discussion by hazirahmokmin** [iTwin/itwinjs-core#5191] [Q&A] [answered]
Hi, I tried to run API to get the form definition based on the project ID but everytime I tried to run it, I faced with a "500 Internal Server Error". I've checked and have been granted the administrator role for all the projects but regardless, it doesn't work. Any way for me to resolve this?
![API_500 Internal Server Error](https://user-images.githubusercontent.com/75367451/224624611-06ac68b6-306f-420b-9485-027e8cda761d.png)

**Comment by pratikshan85** [comment]
@hazirahmokmin can you please provide the request URI and the screenshot of the whole try out operation?

**Reply by arnab-git-hub** [reply]
@hazirahmokmin Can you please provide the information asked by @pratikshan85

**Reply by hazirahmokmin** [reply]
1. cleared authorisation code
![image](https://user-images.githubusercontent.com/75367451/226568017-b7bb63d6-9068-4b2d-8557-c4638f9c6fe8.png)

2. Copied and paste the project ID from SYNCHRO
![image](https://user-images.githubusercontent.com/75367451/226568266-f4fdb32b-355e-4b62-8e8e-fb7c21e96b25.png)

3. after i click execute, the error appeared
![image](https://user-images.githubusercontent.com/75367451/226568358-3f96d265-f03d-4dda-82d2-3ad3efcc3edf.png)

tried with all other project IDs but it still didn't work. also tried with other devices but same issue. however, it works from my colleague's account.

**Reply by arnab-git-hub** [reply]
@pratikshan85

**Comment by pratikshan85** [comment]
@hazirahmokmin , According to your above screenshot you are trying to get the details of a particular Form and there you are passing the id value as projectId instead of passing a valid Form Id. That is the reason you are getting this Internal server error. 

If you want to get the form details based on a ProjectId then please find the below image for the API details :

**GET https://api.bentley.com/forms/?projectId=336fcc3c-eedd-4561-8f1b-387b8e2f8b70**

![image](https://user-images.githubusercontent.com/48123387/228509852-5c308544-bc0a-43f4-8e94-06a2abe0597e.png)

**Reply by hazirahmokmin** [reply]
I've tried using a valid form ID but the same error still appears.

![image](https://user-images.githubusercontent.com/75367451/228695702-77accea2-a962-430f-aaf5-6ceb588c58c0.png)
![image](https://user-images.githubusercontent.com/75367451/228695718-2bb37818-a5e3-429f-b830-dda5d58d31fc.png)

It's odd that when my colleague tried to carry out the same steps using the projectID as per my initial steps, he has no issue doing it from his account. But this is only happening for mine. Any resolution for this?

**Reply by pratikshan85** [reply]
@hazirahmokmin  Can you please provide us your current/recent token to dig further to get the exact reason for the error ?
But it looks like from your example the form Id is not in correct format.
Example of form id :  **"PMxvM93uYUWPGzh7ji-LcFIW9Vv8H4RHsYYV6EuoVi0",**

**Reply by hazirahmokmin** [reply]
sorry, how do I retrieve this token information?

**Reply by pratikshan85** [reply]
@hazirahmokmin Please provide the correct Form Id to the API as I have mentioned in my previous post ( Check the format).
@hazirahmokmin  Once you call the API from the developer portal you can see the Authorization code (Token) in UI .
Please refer to the image below :

![image](https://user-images.githubusercontent.com/48123387/229057181-e63ff8b6-3353-46bd-8287-6efa2c89e32a.png)

**Reply by hazirahmokmin** [reply]
i extracted the formID from the highlighted portion in my web browser. Is this not the right one?
![image](https://user-images.githubusercontent.com/75367451/229446165-2ced0be7-5681-4cf0-9642-cd80690a8de3.png)

Anyway, please find following the token used.
Bearer eyJhbGciOiJSUzI1NiIsImtpZCI6IkJlbnRsZXlJTVNfMjAyMyIsInBpLmF0bSI6ImE4bWUifQ.eyJzY29wZSI6WyJmb3JtczpyZWFkIiwiZm9ybXM6bW9kaWZ5Il0sImNsaWVudF9pZCI6Iml0d2luLWRldmVsb3Blci1jb25zb2xlIiwiYXVkIjpbImh0dHBzOi8vaW1zLmJlbnRsZXkuY29tL2FzL3Rva2VuLm9hdXRoMiIsImh0dHBzOi8vaW1zb2lkYy5iZW50bGV5LmNvbS9hcy90b2tlbi5vYXV0aDIiLCJodHRwczovL2ltc29pZGMuYmVudGxleS5jb20vcmVzb3VyY2VzIiwiYmVudGxleS1hcGktbWFuYWdlbWVudCJdLCJzdWIiOiIxNGRmNTNlMC1kZTk5LTRkNWQtYmZiOS1iMzUwNzAxZjZlNjMiLCJyb2xlIjpbIlNJVEVfQURNSU5JU1RSQVRPUiIsIlByb2plY3QgTWFuYWdlciJdLCJvcmciOiJhMjYwMTg3NS1hODc0LTRmZWMtOWY3My0yMGQxZjRmZTg0NTMiLCJzdWJqZWN0IjoiMTRkZjUzZTAtZGU5OS00ZDVkLWJmYjktYjM1MDcwMWY2ZTYzIiwiaXNzIjoiaHR0cHM6Ly9pbXMuYmVudGxleS5jb20iLCJlbnRpdGxlbWVudCI6WyJNQU5BR0VEX1NFUlZJQ0VTIiwiU0VMRUNUIl0sInByZWZlcnJlZF91c2VybmFtZSI6InNpdGlfbnVyaGF6aXJhaF9tb2ttaW5Ab3B0aW11cy1wdy5jb20iLCJnaXZlbl9uYW1lIjoiU2l0aSIsInNpZCI6Ik90d2NDdDdEclRUazYwVEFINnNLeDNFYTM5US5TVTFUTFVKbGJuUnNaWGt0VTBjLktoeUIuamFTamJBUWM2NGp0bWJ3OVlpdXJPeGRMNCIsIm5iZiI6MTY4MDUwODA0MSwidWx0aW1hdGVfc2l0ZSI6IjEwMDQwODU0NzkiLCJ1c2FnZV9jb3VudHJ5X2lzbyI6IlNHIiwiYXV0aF90aW1lIjoxNjgwNTA4MzQxLCJuYW1lIjoic2l0aV9udXJoYXppcmFoX21va21pbkBvcHRpbXVzLXB3LmNvbSIsIm9yZ19uYW1lIjoiSlRDIENvcnBvcmF0aW9uIiwiZmFtaWx5X25hbWUiOiJOdXJoYXppcmFoIE1va21pbiIsImVtYWlsIjoic2l0aV9udXJoYXppcmFoX21va21pbkBvcHRpbXVzLXB3LmNvbSIsImV4cCI6MTY4MDUxMTk1MH0.jqzElQp3wfziQYZQj5W_KEg4yFGLOY-pKROVNXi-C89r_J-BQkmsuKfeTuEVlQKPJtyNTwgeF3723p26Y7B4KPxVlAdkjH7EoFAtqWhEzN9g4cCptljdi3G0A3TflWJoElKq8HAv3mGR7rNnUPVpYYMzKnhAEBk2iJFiwKAO_pgwYuCiQ-Zq8g2n5RZhhjcF0IwmdBtS0mwgudbFvFir6ykq_oaEhC90wk-9IKJAEHmhTi5dQ_nWTWEDQ5EGugOAWsBR4l0lsGuNxqnqafHujSKx0Yyww_qqRKHYmHxxploGBPBhfkdcMitwLRkpwt3HCLM-eJsXvIU3b7f24UBmdQ

**Comment by pratikshan85** [comment]
@hazirahmokmin The id you are providing to the API is not the correct form Id.
Please call the below API to get all the available Form in a particular project and take one of the Form Id from the list to get it's deatils.
GET https://api.bentley.com/forms/formDefinitions?projectId[&type]

Please check the below link for more information.
https://developer.bentley.com/apis/forms/operations/get-project-form-definitions/

**Reply by hazirahmokmin** [reply]
I've tried but it didn't work.
Have copied the project ID as below from the URL.
![image](https://user-images.githubusercontent.com/75367451/229957180-9f87082b-027b-4e73-a558-28db8c7d0553.png)

After completing the authorisation and pasting the project ID in the "Id" field, the same error appears. I followed the exact same steps using my colleague's account and it worked. Is there an issue with my account? I've checked my access to the project and I do have admin access.
![image](https://user-images.githubusercontent.com/75367451/229957293-3728faf0-b52d-447c-9e7f-6eb3148bcbdf.png)
![image](https://user-images.githubusercontent.com/75367451/229957301-547ee8a6-6d32-43d5-8de3-e250f9b21f3c.png)
![image](https://user-images.githubusercontent.com/75367451/229957320-38f4a68f-a0f1-4148-8562-5d3b01a2ed64.png)

**Reply by pratikshan85** [reply, marked as answer]
@hazirahmokmin 
You are using the wrong API to get the form definition under a project you need to use the below api 
GET https://api.bentley.com/forms/formDefinitions?projectId[&type] 

Please check the below image 

![image](https://user-images.githubusercontent.com/48123387/230052339-0e8ff7d8-51a6-48b0-af47-ae9ec449406d.png)

The one you are using according to your image is you are trying to get details of a particular form , so there you need to pass
the for definition id not the project id.
